resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -241,6 +244,56 @@ class MousePointer {
     }
 }
 
+[DSCResource()]
+class Scrollbar {
+
+    [DscProperty(Key)] [Bool] $Show = $false
+
+
+
+    hidden [string] $DynamicScrollbarProperty = 'DynamicScrollbar'
+
+    [Scrollbar] Get() {
+
+        $currentState = [AlwaysShowScrollbar]::new()
+        
+		if (-not(DoesRegistryKeyPropertyExist -Path $global:ControlPanelAccessibilityRegistryPath -Name $this.DynamicScrollbarProperty)) {
+            $currentState.Show = $false        
+        } else {
+			$dynamicScrollbarValue = (Get-ItemProperty -Path $global:ControlPanelAccessibilityRegistryPath -Name $this.DynamicScrollbarProperty).DynamicScrollbar
+			$currentState.Show = ($dynamicScrollbarValue -eq 0) ? $true : $false
+
+				
+		}
+		
+        return $currentState
+    }
+
+    [bool] Test() {
+        $currentState = $this.Get()
+        if ($this.Show -ne $currentState.Show) {
+            return $false
+        }
+
+
+        return $true
+    }
+
+    [void] Set() {
+        if ($this.Test() -ne $true) {        
+            $desiredState = $this.Show ? 0 : 1
+        }
+

```

Comment:
If the Test method is already in its desired state, then we don't need to do anything. you can return early. 

Move the $desiredState line to after this check.

Reply:
This suggestion contradicts a [previous](https://github.com/microsoft/winget-dsc/pull/49#discussion_r1723889072) suggestion.

Reply:
This was an example to show you that you only need to perform an action if the test method returns $false. This was not meant to be the final implementation.